### PR TITLE
Feature/upload user logo

### PR DESCRIPTION
**Pull request by Myroslaw**
Add frontend part of uploading user logo, but in future need special service for uploading this on server.

### CODE REVIEW

File: app/home/home.controller.js [hunk 1]
```diff
@@ -6,11 +6,15 @@
...
-    function HomeController($log) {
+    HomeController.$inject = ['$log', 'AppModalService'];
+    function HomeController($log, AppModalService) {
         const vm = this;
 
-        ////
+
+        vm.UploadUserLogo = AppModalService.UploadUserLogo;
+
+        activate();
```
Comment (oleh-polishchuk) [inline app/home/home.controller.js]: You try to execute function, which not exist. Please, fix it.

File: app/home/home.controller.js [hunk 2]
```diff
@@ -6,11 +6,15 @@
         .module('app')
         .controller('HomeController', HomeController);
 
-    HomeController.$inject = ['$log'];
-    function HomeController($log) {
+    HomeController.$inject = ['$log', 'AppModalService'];
+    function HomeController($log, AppModalService) {
```
Comment (oleh-polishchuk) [inline app/home/home.controller.js]: If you don't use $log dependencies, please remove it.

File: app/home/home.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <div class="ibox-content">
         <div class="row home">
             <div class="col-xs-4 col-sm-3">
-                <img class="img-circle img-responsive center-block m-b-sm" src="static/images/demo/profile_photo.jpg" alt="profile photo">
+                <img class="img-circle img-responsive center-block m-b-sm" src="static/images/demo/profile_photo.jpg" alt="profile photo" ng-click="vm.UploadUserLogo()">
```
Comment (oleh-polishchuk) [inline app/home/home.html]: Please, wrap this image in empty link (`<a hre=""></a>`), so user can undestend, that this image is clickable.
And set max height and max width for image.

File: app/utils/modal.dialog.service.js [hunk 4]
```diff
@@ -25,5 +25,15 @@
...
+                animation: true,
+                templateUrl: 'utils/upload.user.logo/upload.user.logo.modal.html',
+                controller: 'UploadUserLogoController',
+                controllerAs: 'vm',
+                // size: 'lg'
+            })
+        }
     }
-})();
+})();
```
Comment (oleh-polishchuk) [inline app/utils/modal.dialog.service.js]: Don't forget to add one empty line at the end of file.

File: app/utils/upload.user.logo/ulpoad.user.logo.controller.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+        function activate() {
+            $log.debug('Init UploadUserLogoController ...');
+        }
+
+        function close(data) {
+            $uibModalInstance.close(data);
+        }
+    }
+})();
```
Comment (oleh-polishchuk) [inline app/utils/upload.user.logo/ulpoad.user.logo.controller.js]: The same here

File: app/utils/upload.user.logo/upload.user.logo.modal.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+               ng-model="vm.data.image"
+               ng-click="vm.form.$pristine = false">
+        <label for="file" class="btn btn-primary btn-file" ng-switch on="vm.data.image">
+            <span ng-switch-when=''>Виберіть будь-ласка файл</span>
+            <span ng-switch-default>Обрати інший файл</span>
+        </label>
+
+        <img class="item-image" id="image" alt="Item Image"
+             fallback-src="{{ vm.noImageAvailable }}" ng-src="{{ vm.data.image }}">
+
```
Comment (oleh-polishchuk) [inline app/utils/upload.user.logo/upload.user.logo.modal.html]: Please, add Save button to save uploaded logo.

**Comment by Myroslaw** [comment]
[[ui] RR-45 Upload user logo](https://trello.com/c/ko5GS4Ra/19-ui-rr-45-upload-user-logo)

**Comment by Myroslaw** [comment]
Added form data for uploading image, but need page reload
Added in html automatic avatar url and fallback src

**Pull request merged**